Migrations+Triggers

AlexandruBaris opened the pull request.

Commit:
First Commit

Commit:
Created sql script for table pay;
Created sql script for table projects;
Created trigger on inserted in/deleted operations from departments on table locations;
Created sql script for table employment_logs;
Created function insert_logs;
Created trigger on INSERT INTO or DELETE FROM employees on table employment_logs;

codreanvs commented on src/main/resources/db/migration/V20191004032400__create_table_locations.sql:
```diff
@@ -5,7 +5,8 @@ CREATE TABLE locations
     state_province VARCHAR2(25),
-    country_id     CHAR(2) NOT NULL
+    country_id     CHAR(2) NOT NULL,
+    department_amount NUMBER(3)
```
Please create a separated migration to add this new column.

AlexandruBaris replied:
created

Commit:
Created sql script for table pay;
Created sql script for table projects;
Created trigger on inserted in/deleted operations from departments on table locations;
Created sql script for table employment_logs;
Created function insert_logs;
Created trigger on INSERT INTO or DELETE FROM employees on table employment_logs;

Commit:
refactored

vermurachi commented on homework-selects/sql_questions.sql:
```diff
@@ -3,56 +3,419 @@
+SELECT A.first_name, A.last_name, B.department_name, C.city, C.state_province
+FROM employees A
+         JOIN DEPARTMENTS B
+            ON A.DEPARTMENT_ID = B.DEPARTMENT_ID
```
You could use LEFT JOIN ... USING like in the first example to make the commands shorter

vermurachi replied:
You could use LEFT JOIN ... USING like in the first example to make the commands shorter

vermurachi commented on homework-selects/sql_questions.sql:
```diff
@@ -3,56 +3,419 @@
 --11. the name of the department, average salary and number of employees working in that department who got commission.
+SELECT A.DEPARTMENT_NAME, AVG(B.SALARY),COUNT(COMMISSION_PCT)
+FROM EMPLOYEES B
+         JOIN DEPARTMENTS A ON B.DEPARTMENT_ID = A.DEPARTMENT_ID
```
where do you check the employees who got commission?

vermurachi replied:
where do you check the employees who got commission?

vermurachi commented on homework-selects/sql_questions.sql:
```diff
@@ -3,56 +3,419 @@
+) AND DEPARTMENT_ID IN (
+    SELECT DEPARTMENT_ID
+    FROM EMPLOYEES
+    WHERE FIRST_NAME LIKE '%J'
```
This will select only the names that ends with 'J'

AlexandruBaris replied:
changed %J%

vermurachi commented on homework-selects/sql_questions.sql:
```diff
@@ -3,56 +3,419 @@
+) AND DEPARTMENT_ID IN (
+    SELECT DEPARTMENT_ID
+    FROM EMPLOYEES
+    WHERE FIRST_NAME LIKE '%J'
```
This will select only the names that ends with 'J'

vermurachi commented on homework-selects/sql_questions.sql:
```diff
@@ -3,56 +3,419 @@
+SELECT FIRST_NAME, LAST_NAME, SALARY, DEPARTMENT_ID
+FROM EMPLOYEES
+WHERE SALARY > ANY(
+    SELECT SALARY
```
Where do you check what is the minimum salary?

vermurachi commented on homework-selects/sql_questions.sql:
```diff
@@ -3,56 +3,419 @@
+SELECT FIRST_NAME, LAST_NAME, SALARY, DEPARTMENT_ID
+FROM EMPLOYEES
+WHERE SALARY > ANY(
+    SELECT SALARY
```
Where do you check what is the minimum salary?

AlexandruBaris replied:
MIN ( SALARY)

vermurachi commented:
Try to keep SQL commands in UPPERCASE and table_names and column_names in lower for readability

Commit:
refactored

Commit:
refactored SQL commands in UpperCase and table_names and column_names in lower for readability
